lib/matplotlib/backends/windowing.py

```diff
@@ -19,9 +21,8 @@ def GetForegroundWindow():
     def SetForegroundWindow(hwnd):
         pass
 
-class FocusManager(object):
-    def __init__(self):
-        self._shellWindow = GetForegroundWindow()
-
-    def __del__(self):
-        SetForegroundWindow(self._shellWindow)
+@contextmanager
```

Comment:
Like it or not FocusManager is public API and needs to go through a deprecation cycle, deprecation note, yada yada.

On the other hand, I am not convinced the new manager needs to be public API.  In fact I'd just deprecate the whole windowing module and move focus_manager into `_backend_tk`, where everything is private...

Reply:
Is it really public API?  It's only ever used in `_backend_tk`, and `FocusManager` is entirely undocumented.  The `windowing` module itself is undocumented!

I'll keep it if you insist, but my inclination would be to delete the module in favor of a private context manager...